src/eduid_common/api/schemas/csrf.py

```diff
@@ -15,6 +16,25 @@ class CSRFRequestMixin(Schema):
 
     @validates('csrf_token')
     def validate_csrf_token(self, value):
+        custom_header = request.headers.get('X-Requested-With', '')
+        if custom_header != 'XMLHttpRequest':
+            raise ValidationError('CSRF missing custom X-Requested-With header')
+        origin = request.headers.get('Origin', None)
+        if origin is not None:
+            origin = origin.split()[0]
```

Comment:
Why is it necessary to split and get the first index for the Origin, but not for the Referer header?

Reply:
[Here](https://wiki.mozilla.org/Security/Origin) (referenced from the [OWASP article](https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)_Prevention_Cheat_Sheet#Checking_the_Origin_Header)) they say that the Origin header generally has format: "Origin: <origin> [<origin>]*".
However, in MDN [they say](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin) that it has format "Origin: <scheme> "://" <hostname> [ ":" <port> ]". So, just to be on the safe side, I am taking the first entry in case it s a list.

On the other hand, I've never seen a Referer with a list of URLs, or seen it described as such...

In any case it cannot do any harm, (no chance of getting spaces before the hostname of the first URL), so I'll move